test/Array/array_sort.js

```diff
@@ -134,6 +134,35 @@ const tests = [
         }
     },
     {
+        name : "Array.prototype.sort with edited prototypes and compare function side effects",
```

Comment:
This test had to be revised as the behaviour it was testing was wrong after the spec change.

ALSO had to move it up above other tests that edit the prototype.